tests/t0099_parsing.apln

```diff
@@ -333,41 +333,51 @@ ERRS←{⍺←⊢ ⋄ ⍵⍵::1 ⋄ 0⊣⍺ ⍺⍺ ⍵}
  CHK lx=4 0 0 1 1 1 1 1 1 1 1 1
  CHK pos end=(0 11 12 14 17 20 26 29 34 37 39 41)(59 45 43 16 19 22 27 30 35 38 40 42)
  CHK xn xt≡¨(,⊂'FN')(,4)
- 
+
  CHK PARSE NS '∇FN' 'L1:' '∇'
  CHK t=CD.(F T H L)
  CHK p d=(0 0 1 1)(0 1 2 2)
  CHK k=0 1 0 0
  CHK n=0 (-sym⍳⊂'FN') 0(-sym⍳⊂'L1')
  CHK lx=4 0 0 0
  CHK pos end=(0 11 12 15)(34 20 14 17)
- 
+
  CHK PARSE NS '∇FN' ':If 0' ':EndIf' '∇'
- 
- CHK (,⊂'⍞SOUND')(,⊂2)TOKENIZE NS '∇FN' '⍞SOUND 5' '∇' 
+
+ CHK (,⊂'⍞SOUND')(,⊂2)TOKENIZE NS '∇FN' '⍞SOUND 5' '∇'
  CHK PARSE NS '('':namespace'')' ''
  CHK PARSE NS '∇FN;⎕IO' '∇'
  CHK TOKENIZE NS'∇FN' '{5:6}⍬' '∇'
  CHK TOKENIZE ,⊂'1 ¯2'
  CHK TOKENIZE ERRS 2 ,⊂'1¯2'
  CHK TOKENIZE ERRS 2 ,⊂'1.¯2'
- 
+
  CHK PARSE NS 'FN←{0:5}' ''
  CHK t=CD.(F B V C F G A A F G A A)
  CHK p d=(0 0 1 1 3 4 5 5 3 8 9 9)(0 1 2 2 3 4 5 5 3 4 5 5)
  CHK k=0 2 2 2 2 0 1 1 3 0 1 1
- 
+
  CHK PARSE NS 'FN←{0::5}' ''
  CHK t=CD.(F B V C F G A A F G A A)
- 
+
  CHK PARSE NS ⊂'TEST←{F←{⍵} ⋄ F X←5}'
  CHK t=CD.(F B V C F B V C F P F P B A V V A F B V C F P F P B A V V A)
  CHK p=0 0 1 1 3 4 5 5 7 8 7 10 4 12 13 13 12 3 17 18 18 20 21 20 23 17 25 26 26 25
  CHK d=0 1 2 2 3 4 5 5 6 7 6 7 4 5 6 6 5 3 4 5 5 6 7 6 7 4 5 6 6 5
  CHK k=0 2 2 2 2 2 2 2 2 1 3 1 1 7 1 1 1 3 2 2 2 2 1 3 1 1 7 1 1 1
- 
- ⎕←'☺' 
- 
+
+ CHK PARSE NS 'fn←⊢' 'a←1' '∇FN' 'a fn b←1' '∇'
+ CHK t=CD.(F B V P B V A T H E V V B V A)
+ CHK p=0 0 1 1 0 4 4 0 7 7 9 9 9 12 12
+ CHK k=0 2 2 2 1 1 1 1 0 2 1 2 1 1 1
+
+ CHK PARSE NS 'fn←⊢' 'a←1' 'FN←{a fn b←1}'
+ CHK t=CD.(F B V P B V A B V C F B A V V V A F B A V V V A)
+ CHK p=0 0 1 1 0 4 4 0 7 7 9 10 11 12 12 12 11 9 17 18 19 19 19 18
+ CHK k=0 2 2 2 1 1 1 2 2 2 2 1 7 1 1 1 1 3 1 7 1 1 1 1
+
+ ⎕←'☺'
+
```

Comment:
I tried to make these minimally dependent on unrelated faculties---like scoped bindings---but please sanity check. In particular `a` is factored out to the top level to circumvent dependence on scoped binding lookup.